### ISSUE TITLE
Demo for turn based grid game

### ISSUE THREAD

**Issue by CowThing** [godotengine/godot-demo-projects#13]
Roguelike or board game. This would be a useful demo to have.

**Comment by aaronfranke** [comment]
Is this still desired? We currently have a few grid-based demos already.

What are some minimal game ideas we could use for this? Perhaps chess?

**Comment by CowThing** [comment]
Yeah at the time that I made this issue there weren't many grid-based examples for Godot. And some of the examples I found from other engines used some strange methods, like using the physics engine and ray casts to check grid positions.
I do think that a turn-based demo would still be useful. Chess would be good because it can show grid-based gameplay, pieces that have different rules, and turn based gameplay where each player can only interact with their pieces on their turn.

**Comment by InfiniteProductions** [comment]
@CowThing  wouldn't be easier in a purpose of a demo to make something like a single player _Tic Tac Toe_ game ?

I understand the main purpose is to generate a grid on screen which have configurable size for cells and columns & rows, then put something in the right screen coordinates according to logical location inside the grid.

Alternate turns and pieces specific rules are a bit out of scope of such demo IMHO.

If so, I made everything needed to make a board game (it's an _Ishido - The Way of Stones_ clone) and I could take some time to extract base logic, still only tested so far using fixed size window, so there may be some surprise on resolution change.
There is also the nice [Match3 tutorial from Mister Taft Creates](https://www.youtube.com/watch?v=RO5pXiN6UnI&list=PL4vbr3u7UKWqwQlvwvgNcgDL1p_3hcNn2&index=2) which also do some grid <-> screen conversion calculations.
